plugins/orchestrator-backend/src/service/SonataFlowService.ts

```diff
@@ -180,12 +190,114 @@ export class SonataFlowService {
         );
         return items.filter((item): item is WorkflowItem => !!item);
       }
+      this.logger.error(
+        `Response was NOT okay when fetch(${this.url}/management/processes). Received response: ${response}`,
+      );
     } catch (error) {
       this.logger.error(`Error when fetching workflows: ${error}`);
     }
     return undefined;
   }
 
+  public async fetchWorkflowOverviews(): Promise<
+    WorkflowOverview[] | undefined
+  > {
+    try {
+      const response = await executeWithRetry(() =>
+        fetch(`${this.url}/management/processes`),
+      );
+
+      if (response.ok) {
+        const workflowIds = (await response.json()) as string[];
+        if (!workflowIds?.length) {
+          return [];
+        }
+        const items = await Promise.all(
+          workflowIds.map(async (workflowId: string) => {
+            const definition = await this.fetchWorkflowDefinition(workflowId);
```

Comment:
Maybe worth separating the callback function of workflowIds.map to 
getWorkflowOverview
so you can reuse it for the overview API for a single workflow
/api/orchestrator/workflows/:id/overview

Reply:
[245109b](https://github.com/caponetto/backstage-plugins/pull/36/commits/245109bbd624d98be3dd7da6883925882b996ca6)